perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -5532,8 +5533,21 @@ public void deleteGroupsFromAutoRegistration(PerunSession sess, List<Group> grou
 	}
 
 	@Override
-	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) {
+	public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) throws GroupNotAllowedToAutoRegistrationException {
 		for (Group group : groups) {
+			if (group.getName().equals(VosManager.MEMBERS_GROUP)) {
+				throw new GroupNotAllowedToAutoRegistrationException("Members group cannot be added to auto registration.", group);
+			}
+
+			try {
+				Attribute syncEnabledAttr = getPerunBl().getAttributesManagerBl().getAttribute(sess, group, GroupsManager.GROUPSYNCHROENABLED_ATTRNAME);
+				if (syncEnabledAttr.valueAsString() != null && syncEnabledAttr.valueAsString().equals("true")) {
```

Comment:
This could be simplified: `if ("true".equals(syncEnabledAttr.valueAsString())) {`

Reply:
Fixed.